### PR TITLE
S3t1 user classes

### PR DESCRIPTION
**Pull request by nesclark**

### CODE REVIEW

File: src/Hydrators/UserHydrator.php [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    /**
+     * retrieves user data from db
+     * @param \PDO $db
+     * @param int $id
+     * @return mixed
+     */
+    public static function getUserById(\PDO $db, int $id)
+    {
+        $stmnt = $db->prepare("SELECT * FROM  project_users LEFT JOIN users ON project_users.`user_id` = users.`id` WHERE project_users.`project_id` = :id;");
+        $stmnt->setFetchMode(PDO::FETCH_CLASS, UserEntity::class);
```
Comment (Steph-ski) [inline src/Hydrators/UserHydrator.php]: add in a \ before PDO::

File: src/Hydrators/UserHydrator.php [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+    /**
+     * retrieves user data from db
+     * @param \PDO $db
+     * @param int $id
+     * @return mixed
+     */
+    public static function getUserById(\PDO $db, int $id)
+    {
+        $stmnt = $db->prepare("SELECT * FROM  project_users LEFT JOIN users ON project_users.`user_id` = users.`id` WHERE project_users.`project_id` = :id;");
```
Comment (Steph-ski) [inline src/Hydrators/UserHydrator.php]: should the * in the SELECT statement be replaced with all of the UserEntity properties?

File: src/Entities/ProjectEntity.php [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    public function jsonSerialize(): array
+    {
+        return [
+            'id' => $this->id,
+            'name' => $this->name,
+            'client_id' => $this->client_id,
+            'deadline' => $this->deadline,
+            'overdue' => $this->calculateOverdue()
+        ];
+
```
Comment (Steph-ski) [inline src/Entities/ProjectEntity.php]: remove additional space?

File: src/Entities/ProjectEntity.php [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    }
+
```
Comment (Steph-ski) [inline src/Entities/ProjectEntity.php]: remove additional space?

File: src/Entities/UserEntity.php [hunk 5]
```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace ProjMange\Entities;
+
+class UserEntity
```
Comment (Steph-ski) [inline src/Entities/UserEntity.php]: implements JsonSerializable - needs to be added in after class UserEntity

File: src/Hydrators/UserHydrator.php [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+namespace ProjMange\Hydrators;
+
+use ProjMange\Entities\UserEntity;
+
+class UserHydrator
+{
+
+    /**
+     * retrieves user data from db
+     * @param \PDO $db
```
Comment (Steph-ski) [inline src/Hydrators/UserHydrator.php]: think you can remove the \ before PDO here as its not actually being used in the code, its just for description purposes.

File: src/Hydrators/UserHydrator.php (hunk 2 above)
Comment (alreuben) [reply src/Hydrators/UserHydrator.php]: replace with individual fields instead of selecting everything

**Review by Steph-ski — CHANGES_REQUESTED** [review]
Looks good, couple of small comments which need reviewing.

**Comment by randallcoding** [comment]
Looks good, what could be covered seems to have already been covered by Steph

**Pull request merged**